tests/NuGet.Services.BasicSearchTests/TestSupport/LuceneInitializer.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Lucene.Net.Store;
+using NuGet.Indexing;
+
+namespace NuGet.Services.BasicSearchTests.TestSupport
+{
+    public class LuceneInitializer
+    {
+        private readonly BasicSearchTestSettings _settings;
+        private readonly DataServicePackageRepository _repository;
+
+        public LuceneInitializer(BasicSearchTestSettings settings)
+        {
+            _settings = settings;
+            _repository = new DataServicePackageRepository(_settings.PackageSource);
+        }
+
+        public async Task InitializeAsync()
+        {
+            await DownloadPackagesAsync();
+            CreateLuceneIndex();
+        }
+
+        private async Task DownloadPackagesAsync()
+        {
+            System.IO.Directory.CreateDirectory(_settings.PackageDirectory);
+            await Task.WhenAll(_settings.PackageVersions.Select(DownloadPackageAsync));
+        }
+
+        private void CreateLuceneIndex()
+        {
+            var directoryInfo = new DirectoryInfo(_settings.LuceneDirectory);
+            directoryInfo.Create();
```

Comment:
This will required a lot of changes to the service (since it only uses a file system directory right now). I would like to do this as a separate PR, if that's okay.